arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1774,19 +1797,21 @@ int ClusterInfo::createCollectionCoordinator(
   if (numberOfShards == 0 || isSmart) {
     loadCurrent();
     events::CreateCollection(name, TRI_ERROR_NO_ERROR);
-    return TRI_ERROR_NO_ERROR;
+
+    return Result(TRI_ERROR_NO_ERROR);
   }
 
   {
     while (true) {
       int tmpRes = dbServerResult->load(std::memory_order_acquire);
+
       if (tmpRes >= 0) {
         CONDITION_LOCKER(locker, agencyCallback->_cv);
         cbGuard.fire();  // unregister cb before accessing errMsg
-        errorMsg = *errMsg;
         loadCurrent();
         events::CreateCollection(name, *dbServerResult);
-        return tmpRes;
+
+        return Result(tmpRes, *errMsg);
```

Comment:
is `errMsg` still going to be populated here, or can it be removed?

Reply:
that's the string that gets populated in the asynchronous task lines 1658, 1663, its value gets returned as part of the function result